### DISCUSSION TITLE
Sequential SOS

### DISCUSSION THREAD

**Discussion by JOlucak** [iFR-ACSO/casos#67] [Ideas]
A first implementation of  sequential SOS is available. It works fine and we obtain the same results as in bisos. 

This thread is meant to keep track of certain known issues.

**Comment by JOlucak** [comment]
- [x] Different results for GTM 4D example (results (level set) is worse, linesearch does not work. Check where the difference come from.
- [x] Improve building-time i.e. more efficient implementation
- [x] Check why mosek throws an error during the linesearch
- [x] Clean up constructor and move code to buildproblem
- [ ] Look and implement on more sophisticated line-search algorithms

**Comment by tcunis** [comment]
[ ] Handle Gram decision variables

**Comment by JOlucak** [comment]
- [x] Adapt line search algorithm (probably low level solver)

**Comment by JOlucak** [comment]
Current status
- The linesearch-solver using Mosek is replaced by a simple bisection
- A verbosity option is added showing the iteration, current cost, difference in primal and dual variables
- Implemented some examples

**Comment by JOlucak** [comment]
Some new issues occured that need to be clarified

- [ ] Cost is non decresing

- [ ] Some constraint are not fulfilled in the reachability problem. The storage function zero level set at t = T i.e. V(T,x) \leq 0 is outside the pre-defined terminal set 

![grafik](https://github.com/iFR-ACSO/casos/assets/61011693/f49c0571-0619-473e-96c6-3271d319bca6)

**Comment by JOlucak** [comment]
Current Status and To-Dos 06.05.2024
- Sequential approach from original paper works
- Extended to quadratic cost and Hessian estimation using BFGS
- First version of filter linesearch implemented (naive implementation)



To Dos:

- [ ] Clean up examples (linear and quadratic)
- [ ] Rename examples
- [ ] Clean up build problem
- [ ] Improve globalization to more sophisticated methods
